set.mm

```diff
@@ -54198,13 +54208,20 @@ Definite description binder (inverted iota)
   $}
 
   ${
-    $d y x $.  $d y ph $.
+    $d x y $.
+    $( Lemma for ~ dfiota2 , ~ dfaiota2 and ~ aiotaval .  Formerly part of
+       proof of ~ dfiota2 .  (Contributed by Andrew Salmon, 30-Jun-2011.)
+       (Revised by AV, 24-Aug-2022.) $)
+    dfiota2lem $p |- ( { x | ph } = { y } <-> A. x ( ph <-> x = y ) ) $=
+      ( cab cv csn wceq weq wb wal df-sn eqeq2i abbi bitr4i ) ABDZCEZFZGOBCHZBD
+      ZGARIBJQSOBPKLARBMN $.
```

Comment:
I would replace `y` with `A`. I think this is more than a lemma. It could be moved earlier with a comment like
> Condition for a class abstraction to be a singleton.

as the label you see fit.

Reply:
OK, I will try - but it is a little bit more complicted, because ` A ` might be a proper class...

Reply:
Done, it was easier than I expected (no case discrimination necessary)